src/Money.php

```diff
@@ -537,6 +540,8 @@ public function allocate(int ...$ratios) : array
             $remainder = $remainder->minus($unit);
         }
 
```

Comment:
No assertions please, I don't use them anywhere else in this library, and test cases are expected to catch the bugs (a test case for negative numbers was missing though, as you found out! 👍 )